elfi/examples/bignk.py

```diff
@@ -0,0 +1,296 @@
+    elfi.Prior('uniform', -5, 10, model=m, name='g2')
+    elfi.Prior('uniform', -.5, 5.5, model=m, name='k1')
+    elfi.Prior('uniform', -.5, 5.5, model=m, name='k2')
+    elfi.Prior('uniform', -1 + EPS, 2 - 2*EPS, model=m, name='rho')
+
+    # Generating the observations.
+    y_obs = BiGNK(*true_params, n_obs=n_obs,
+                  random_state=np.random.RandomState(seed_obs))
+
+    # Defining the simulator.
+    fn_sim = partial(BiGNK, n_obs=n_obs)
+    elfi.Simulator(fn_sim, m['a1'], m['a2'], m['b1'], m['b2'], m['g1'],
+                   m['g2'], m['k1'], m['k2'], m['rho'], observed=y_obs,
+                   name='BiGNK')
+
+    # Initialising the chosen summary statistics.
+    fns_summary_all = [ss_order, ss_robust, ss_octile]
+    fns_summary_chosen = []
+    for fn_summary in fns_summary_all:
+        if fn_summary.__name__ in stats_summary:
+            summary = elfi.Summary(fn_summary, m['BiGNK'],
+                                   name=fn_summary.__name__)
+            fns_summary_chosen.append(summary)
+
+    # Defining the distance metric.
+    elfi.Discrepancy(euclidean_multidim, *fns_summary_chosen, name='d')
+
+    return m
+
+
+def euclidean_multidim(*simulated, observed):
+    """Calculating the multi-dimensional Euclidean distance.
+
+    Parameters
+    ----------
+    *simulated: array_like
+        The simulated points.
+    observed : array_like
+        The observed points.
+
+    Returns
+    -------
+    array_like
+    """
+    pts_sim = np.column_stack(simulated)
+    pts_obs = np.column_stack(observed)
+    d_multidim = np.sum((pts_sim - pts_obs)**2., axis=1)
+    d_squared = np.sum(d_multidim, axis=1)
+    d = np.sqrt(d_squared)
+
+    return d
+
+
+def ss_order(y):
+    """Obtaining the order summary statistic, [2].
+    The statistic reaches an optimal performance upon a low number of
+    observations.
+
+    Parameters
+    ----------
+    y : array_like
+        The yielded points.
+
+    Returns
+    -------
+    array_like
+    """
+    ss_order = np.sort(y)
+
+    return ss_order
+
+
+def ss_robust(y):
+    """Obtaining the robust summary statistic, [1].
+    The statistic reaches an optimal performance upon a high number of
+    observations.
+
+    Parameters
+    ----------
+    y : array_like
+        The yielded points.
+
+    Returns
+    -------
+    array_like
+    """
+    ss_A = _get_ss_A(y)
+    ss_B = _get_ss_B(y)
+    ss_g = _get_ss_g(y)
+    ss_k = _get_ss_k(y)
+
+    ss_robust = np.stack((ss_A, ss_B, ss_g, ss_k), axis=1)
+
+    return ss_robust
+
+
+def ss_octile(y):
+    """Obtaining the octile summary statistic.
+    The statistic reaches an optimal performance upon a high number of
+    observations. As reported in [1], it is more stable than ss_robust.
+
+    Parameters
+    ----------
+    y : array_like
+        The yielded points.
+
+    Returns
+    -------
+    array_like
+    """
+    E1 = ss.scoreatpercentile(y, 12.5, axis=1)
```

Comment:
Most numpy and scipy operations are vectorized. You can simplify this with something like:
```python
arr = np.random.randn(100)
E1, E2, E3, E4, E5 = ss.scoreatpercentile(arr, np.linspace(0, 100, 5))
```